From 0ad52903890fe680307b46f4051bdd91d8aaffd5 Mon Sep 17 00:00:00 2001
From: Eugene Shalygin <eugene.shalygin@gmail.com>
Date: Sat, 7 Oct 2017 21:48:52 +0200
Subject: [PATCH] cmake: do not use Qt5Widgets when locating
 QtSingleApplication. Closes #7551.

This fixes cmake builds with GUI disabled and system
QtSingleApplication. We rely on Qt5::Core instead of Qt5::Widgets.
---
 cmake/Modules/FindQtSingleApplication.cmake | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/cmake/Modules/FindQtSingleApplication.cmake b/cmake/Modules/FindQtSingleApplication.cmake
index eb56607e94..9b526c9839 100644
--- a/cmake/Modules/FindQtSingleApplication.cmake
+++ b/cmake/Modules/FindQtSingleApplication.cmake
@@ -30,9 +30,9 @@ IF(QT4_FOUND)
         NAMES ${QTSINGLEAPPLICATION_NAMES}
         PATHS ${QT_LIBRARY_DIR}
     )
-ELSEIF(Qt5Widgets_FOUND)
+ELSEIF(Qt5Core_FOUND)
     message(STATUS "Looking for Qt5 single application library")
-    FOREACH(TOP_INCLUDE_PATH in ${Qt5Widgets_INCLUDE_DIRS} ${FRAMEWORK_INCLUDE_DIR})
+    FOREACH(TOP_INCLUDE_PATH in ${Qt5Core_INCLUDE_DIRS} ${FRAMEWORK_INCLUDE_DIR})
         FIND_PATH(QTSINGLEAPPLICATION_INCLUDE_DIR QtSingleApplication ${TOP_INCLUDE_PATH}/QtSolutions)
 
         IF(QTSINGLEAPPLICATION_INCLUDE_DIR)
@@ -43,12 +43,12 @@ ELSEIF(Qt5Widgets_FOUND)
     SET(QTSINGLEAPPLICATION_NAMES ${QTSINGLEAPPLICATION_NAMES}
         Qt5Solutions_SingleApplication-2.6 libQt5Solutions_SingleApplication-2.6
         QtSolutions_SingleApplication-2.6 libQtSolutions_SingleApplication-2.6)
-    GET_TARGET_PROPERTY(QT5_WIDGETSLIBRARY Qt5::Widgets LOCATION)
-    GET_FILENAME_COMPONENT(QT5_WIDGETSLIBRARYPATH ${QT5_WIDGETSLIBRARY} PATH)
+    GET_TARGET_PROPERTY(_QT5_CORELIBRARY Qt5::Core LOCATION)
+    GET_FILENAME_COMPONENT(_QT5_CORELIBRARYPATH ${_QT5_CORELIBRARY} PATH)
 
     FIND_LIBRARY(QTSINGLEAPPLICATION_LIBRARY
         NAMES ${QTSINGLEAPPLICATION_NAMES}
-        PATHS ${QT5_WIDGETSLIBRARYPATH}
+        PATHS ${_QT5_CORELIBRARYPATH}
     )
 ENDIF()
 
